js/usercss.js

```diff
@@ -0,0 +1,311 @@
+/* global loadScript mozParser semverCompare */
+
+'use strict';
+
+// eslint-disable-next-line no-var
+var usercss = (function () {
+  const METAS = [
+    'author', 'description', 'homepageURL', 'icon', 'license', 'name',
+    'namespace', 'noframes', 'preprocessor', 'supportURL', 'var', 'version'
+  ];
+
+  const BUILDER = {
+    default: {
+      postprocess(sections, vars) {
+        let varDef = ':root {\n';
+        for (const key of Object.keys(vars)) {
+          varDef += `  --${key}: ${vars[key].value};\n`;
+        }
+        varDef += '}\n';
+
+        for (const section of sections) {
+          section.code = varDef + section.code;
+        }
+      }
+    },
+    stylus: {
+      preprocess(source, vars) {
+        return loadScript('vendor/stylus-lang/stylus.min.js').then(() => (
+          new Promise((resolve, reject) => {
+            let varDef = '';
+            for (const key of Object.keys(vars)) {
+              varDef += `${key} = ${vars[key].value};\n`;
+            }
+
+            // eslint-disable-next-line no-undef
+            stylus(varDef + source).render((err, output) => {
+              if (err) {
+                reject(err);
+              } else {
+                resolve(output);
+              }
+            });
+          })
+        ));
+      }
+    }
+  };
+
+  const colorParser = (function () {
+    const el = document.createElement('div');
+    // https://bugs.webkit.org/show_bug.cgi?id=14563
+    document.head.appendChild(el);
+
+    function _parse(color) {
```

Comment:
ugh, again the underscore. Rename it please to something meaningful like `parseRaw`, parseInternal`, or whatever is actually applicable.